src/content/docs/ja/guides/migrate-to-astro/from-gatsby.mdx

```diff
@@ -0,0 +1,608 @@
+
+#### MarkdownとMDXページ
+
+AstroはMarkdownをネイティブサポートし、MDX用のオプション統合も提供しています。既存の**[MarkdownとMDXファイル](/en/guides/markdown-content/)**を再利用できますが、フロントマターに**[Astro専用の`layout`プロパティ](/en/basics/layouts/#markdown-layouts)**を追加するなどの調整が必要な場合があります。これらのファイルも`src/pages/`に置けば自動的にファイルベースルーティングを利用できます。
+
+あるいはAstroの[コンテンツコレクション](/en/guides/content-collections/)を使ってコンテンツを管理し、自分で取得して[動的にページを生成](/en/guides/content-collections/#generating-routes-from-content)することもできます。
+
+### テストの移行する
+
+Astroは生のHTMLを出力するため、ビルド後の出力を使用したE2Eテストができます。旧Gatsbyサイトのマークアップが再現できれば、既存のエンドツーエンドテストがそのまま動作する場合があります。JestやReact Testing LibraryなどをインポートしてReactコンポーネントをテストできます。
+
+詳しくは[テストガイド](/en/guides/testing/)を参照してください。
+
+### 設定ファイルの再利用する
+
+Gatsbyにはルーティングやメタデータを含むトップレベル設定ファイルが複数あり、ルーティングに使用されます。Astroではこれらの`gatsby-*.js`ファイルを使用しませんが、内容を再利用できる場合があります。
+
+- `gatsby-config.js`: `src/data/siteMetadata.js`（または`siteMetadata.json`）に`siteMetadata: {}`を移動して、ページレイアウトにサイト（タイトル、説明、ソーシャルアカウントなど）に関するデータをインポートします。
+
+- `gatsby-browser.js`: ここで使用されているものは、[主要レイアウト](#migrating-layout-files)の`<head>`タグに直接追加することを考慮してください。
+
+- `gatsby-node.js`: Astroでは独自のノードを作成する必要はありませんが、このファイルのスキーマを確認して、Astroプロジェクトで型を定義するのに役立つかもしれません。
+
+- `gatsby-ssr.js`: AstroでSSRを使用する場合、`astro.config.mjs`に直接[SSRアダプター](/en/guides/on-demand-rendering/)を追加して構成してください。
+
+## 参考: Astro構文への変換する
+
+以下はGatsby特有の構文をAstroに変換するいくつかの例です。Astroコンポーネントの書き方における[AstroとJSXの違い](/en/reference/astro-syntax/#differences-between-astro-and-jsx)も参照してください。
+
+### Gatsby LinksをAstroに変換する
+
+Gatsbyの`<Link to="">`、`<NavLink>`などのコンポーネントをHTMLの`<a href="">`タグに変換します。
+
+```astro del={1} ins={2}
+<Link to="/blog">Blog</Link>
+<a href="/blog">Blog</a>
+```
+
+Astroにはリンク専用のコンポーネントはありませんが、独自の`<Link>`コンポーネントを作ってもかまいません。`<Link>`を他のコンポーネントのようにインポートして使用することができます。
+
+```astro title="src/components/Link.astro"
+---
+const { to } = Astro.props
+---
+<a href={to}><slot /></a>
+```
+
+### Gatsby ImportsをAstroに変換する
+
+必要に応じて、[ファイルのインポート](/en/guides/imports/)を相対ファイルパスで正確に参照するように変更します。これには[インポートエイリアス](/en/guides/typescript/#import-aliases)を使用したり、相対パスを完全に書くこともできます。
+
+注意: `.astro`など一部のファイルは拡張子を省略できません。
+
+```astro title="src/pages/authors/Fred.astro" ".astro"
+---
+import Card from `../../components/Card.astro`;
+---
+<Card />
+```
+
+### Gatsbyの`{children}`をAstroに変換する
+
+Gatsbyの`{children}`をAstroの<slot />に変換します。Astroは`{children}`を関数プロップとして受け取る必要なく、子要素を<slot />で自動的にレンダリングします。
+
+```astro title="src/components/MyComponent" del={3-9} ins={11-13}
+---
+---
+export default function MyComponent(props) { 
+    return (
+      <div>
+        {props.children}
+      </div>
+    );  
+}
+
+<div>
+  <slot />
+</div>
+```
+
+Reactコンポーネントが複数の子要素を渡す場合は、[named slots](/en/basics/astro-components/#named-slots)を使用してAstroコンポーネントに移行できます。
+
+[特定の`<slot />`の使用方法](/en/basics/astro-components/#slots)に関する詳しい説明は[特定の`<slot />`の使用方法](/en/basics/astro-components/#slots)を参照してください。
+
+### スタイリングを変換する
+
+必要に応じて、[CSS-in-JSライブラリ](https://github.com/withastro/astro/issues/4432)（例：styled-components）をAstroで使用できるCSSオプションに置き換える必要があります。
```

Comment:
styled-components as a project is in "maintenance mode". If necessary, delete this line.

https://opencollective.com/styled-components/updates/thank-you